websites/H/hanime/presence.ts

```diff
@@ -1,85 +1,71 @@
-var iFrameVideo: any, currentTime: any, duration: any, paused: any;
-
-presence.on("iFrameData", (data) => {
-  playback = data.iframe_video.dur !== null ? true : false;
+presence.on("iFrameData", (data: { dur: number; currTime: number; paused: boolean; iFrameVideo: boolean} ) => {
+  playback = data.dur !== null ? true : false;
 
   if (playback) {
-    iFrameVideo = data.iframe_video.iFrameVideo;
-    currentTime = data.iframe_video.currTime;
-    duration = data.iframe_video.dur;
-    paused = data.iframe_video.paused;
+    currentTime = data.currTime;
+    duration = data.dur;
+    paused = data.paused;
   }
 });
 
 presence.on("UpdateData", async () => {
-  if (iFrameVideo != false && !isNaN(duration)) {
-    var videoTitle: any, timestamps: any, brand: any;
-
-    videoTitle = document.querySelector(
-      "div > div.title-views.flex.column > h1"
-    );
-    brand = document.querySelector(
-      "div.hvpi-main.flex.column > div > div > div:nth-child(1) > a"
-    );
-    timestamps = getTimestamps(Math.floor(currentTime), Math.floor(duration));
-    presenceData.details =
-      videoTitle !== null ? videoTitle.innerText : "Title not found";
-    presenceData.state = brand.innerText;
-    presenceData.largeImageKey = "logo";
-    presenceData.smallImageKey = paused ? "pause" : "play";
-    presenceData.smallImageText = paused
-      ? (await strings).pause
-      : (await strings).play;
-    presenceData.startTimestamp = timestamps[0];
-    presenceData.endTimestamp = timestamps[1];
-
-    presence.setTrayTitle(
-      paused
-        ? ""
-        : videoTitle !== null
-        ? videoTitle.innerText
-        : "Title not found"
-    );
-
-    if (paused) {
-      delete presenceData.startTimestamp;
-      delete presenceData.endTimestamp;
-    }
-
-    if (iFrameVideo && videoTitle !== null) {
-      presence.setActivity(presenceData, !paused);
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  };
+  if (document.location.pathname.includes("/videos")) {
+    if (playback == true && !isNaN(duration)) {
+      const videoTitle = document.querySelector(
+        "div > div.title-views.flex.column > h1"
+      ),
+       brand = document.querySelector(
+        "div.hvpi-main.flex.column > div > div > div:nth-child(1) > a"
+      ),
+       timestamps = presence.getTimestamps(Math.floor(currentTime), Math.floor(duration));
+      presenceData.details =
+        videoTitle !== null ? videoTitle.textContent : "Title not found";
+      presenceData.state = brand.textContent;
+      presenceData.smallImageKey = paused ? "pause" : "play";
+      presenceData.smallImageText = paused
+        ? (await strings).pause
+        : (await strings).play;
+      presenceData.startTimestamp = timestamps[0];
+      presenceData.endTimestamp = timestamps[1];
+  
+      presence.setTrayTitle(
+        paused
+          ? ""
+          : videoTitle !== null
+          ? videoTitle.textContent
+          : "Title not found"
+      );
```

Comment:
you shouldn't set anything if you don't have anything to set for macOS tray.

Reply:
yeah OK, was debating removing this but left it in. Will remove since it's from the older versions anyway